packages/snaps-simulation/src/structs.test.tsx

```diff
@@ -358,3 +360,60 @@ describe('SnapResponseStruct', () => {
     expect(() => create(value, SnapResponseStruct)).toThrow();
   });
 });
+
+describe('BaseNameLookupOptionsStruct', () => {
+  it('accepts a valid object', () => {
+    const options = create(
+      {
+        chainId: 'eip155:1',
+      },
+      BaseNameLookupOptionsStruct,
+    );
+
+    expect(options).toStrictEqual({
+      chainId: 'eip155:1',
+    });
+  });
+
+  it.each(INVALID_VALUES)('throws for invalid value: %p', (value) => {
+    // eslint-disable-next-line jest/require-to-throw-message
+    expect(() => create(value, BaseNameLookupOptionsStruct)).toThrow();
+  });
+});
+
+describe('NameLookupOptionsStruct', () => {
+  it('accepts a valid object for domain lookup', () => {
+    const options = create(
+      {
+        chainId: 'eip155:1',
+        domain: 'test.domain',
+      },
+      NameLookupOptionsStruct,
+    );
+
+    expect(options).toStrictEqual({
+      chainId: 'eip155:1',
+      domain: 'test.domain',
+    });
+  });
+
+  it('accepts a valid object for address lookup', () => {
+    const options = create(
+      {
+        chainId: 'eip155:1',
+        address: '0xc0ffee254729296a45a3885639AC7E10F9d54979',
+      },
+      NameLookupOptionsStruct,
+    );
+
+    expect(options).toStrictEqual({
+      chainId: 'eip155:1',
+      address: '0xc0ffee254729296a45a3885639AC7E10F9d54979',
+    });
+  });
+
+  it.each(INVALID_VALUES)('throws for invalid value: %p', (value) => {
```

Comment:
Can we test that it throws if you try to use both `domain` and `address`?

Reply:
Added.